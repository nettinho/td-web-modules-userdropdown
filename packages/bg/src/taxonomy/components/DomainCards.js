import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Card, Header, Icon, Message, Divider } from "semantic-ui-react";
import { compose } from "redux";
import { connect } from "react-redux";
import { FormattedMessage, injectIntl } from "react-intl";
import { linkTo } from "../routes";
import { getVisibleDomains } from "../selectors";

const toDomainPath = linkTo.DOMAIN;

export const DomainCard = ({ id, name, type, description, childCount }) => (
  <Card key={id} link as={Link} to={toDomainPath({ id })}>
    <Card.Content>
      <Card.Header>
        <Icon name="cube" />
        {name}
      </Card.Header>
      <Card.Meta>
        <span>{type || <FormattedMessage id="domain" />}</span>
      </Card.Meta>
    </Card.Content>
    <Card.Content description={description} />
    <Card.Content extra>
      <Icon name="cubes" />
      <FormattedMessage
        id="domain.props.children"
        values={{ count: childCount }}
      />
    </Card.Content>
  </Card>
);

DomainCard.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  type: PropTypes.string,
  childCount: PropTypes.number,
  description: PropTypes.string
};

export const DomainCards = ({
  visibleDomains,
  domainsFilter,
  domain,
  intl: { formatMessage }
}) => (
  <React.Fragment>
    <Divider hidden />
    {domainsFilter && (
      <Header as="h4">
        <Icon name="search" />
        <Header.Content>
          {visibleDomains.length == 0 && (
            <FormattedMessage id="domains.search.results.empty" />
          )}
          {visibleDomains.length > 0 && (
            <FormattedMessage
              id="domains.search.results.count"
              values={{ count: visibleDomains.length }}
            />
          )}
        </Header.Content>
      </Header>
    )}
    {_.isEmpty(visibleDomains) &&
      domain &&
      domain.id && (
        <Message
          icon="warning"
          header={formatMessage({ id: "domain.children.empty" })}
          list={[
            <Link key={1} to={linkTo.DOMAIN_NEW({ id: domain.id })}>
              <FormattedMessage id="domain.actions.create" />
            </Link>
          ]}
        />
      )}
    <Card.Group>{visibleDomains.map(DomainCard)}</Card.Group>
  </React.Fragment>
);

DomainCards.propTypes = {
  visibleDomains: PropTypes.array,
  domainsFilter: PropTypes.string,
  domain: PropTypes.object,
  intl: PropTypes.object
};

const mapStateToProps = state => ({
  visibleDomains: getVisibleDomains(state),
  domainsLoading: state.domainsLoading,
  domainsFilter: state.domainsFilter,
  domain: state.domain
});

export default compose(
  injectIntl,
  connect(mapStateToProps)
)(DomainCards);
