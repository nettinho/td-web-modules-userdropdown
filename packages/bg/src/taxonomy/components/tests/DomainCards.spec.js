import React from "react";
import { shallowWithIntl } from "intl-stub";
import { DomainCards } from "../DomainCards";

describe("<DomainCards />", () => {
  const visibleDomains = [
    { id: 1, name: "domain1" },
    { id: 2, name: "domain2" }
  ];
  const domainsFilter = "d";
  const props = { visibleDomains, domainsFilter };

  it("matches the latest snapshot", () => {
    const wrapper = shallowWithIntl(<DomainCards {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
