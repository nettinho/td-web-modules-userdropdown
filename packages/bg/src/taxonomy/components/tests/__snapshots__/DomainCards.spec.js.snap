// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DomainCards /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Divider
    hidden={true}
  />
  <Header
    as="h4"
  >
    <Icon
      as="i"
      name="search"
    />
    <HeaderContent>
      <FormattedMessage
        id="domains.search.results.count"
        values={
          Object {
            "count": 2,
          }
        }
      />
    </HeaderContent>
  </Header>
  <CardGroup>
    <Card
      as={[Function]}
      key="1"
      link={true}
      to="/domains/1"
    >
      <CardContent>
        <CardHeader>
          <Icon
            as="i"
            name="cube"
          />
          domain1
        </CardHeader>
        <CardMeta>
          <span>
            <FormattedMessage
              id="domain"
              values={Object {}}
            />
          </span>
        </CardMeta>
      </CardContent>
      <CardContent />
      <CardContent
        extra={true}
      >
        <Icon
          as="i"
          name="cubes"
        />
        <FormattedMessage
          id="domain.props.children"
          values={
            Object {
              "count": undefined,
            }
          }
        />
      </CardContent>
    </Card>
    <Card
      as={[Function]}
      key="2"
      link={true}
      to="/domains/2"
    >
      <CardContent>
        <CardHeader>
          <Icon
            as="i"
            name="cube"
          />
          domain2
        </CardHeader>
        <CardMeta>
          <span>
            <FormattedMessage
              id="domain"
              values={Object {}}
            />
          </span>
        </CardMeta>
      </CardContent>
      <CardContent />
      <CardContent
        extra={true}
      >
        <Icon
          as="i"
          name="cubes"
        />
        <FormattedMessage
          id="domain.props.children"
          values={
            Object {
              "count": undefined,
            }
          }
        />
      </CardContent>
    </Card>
  </CardGroup>
</React.Fragment>
`;
